simulator/simulator.go

```diff
@@ -65,10 +56,11 @@ func startSimulator() error {
 		return xerrors.Errorf("create an etcd client: %w", err)
 	}
 
-	// need to sleep here to make all controllers create initial resources. (like "system-" priorityclass.)
-	time.Sleep(1 * time.Second)
+	if _, err := client.CoreV1().Namespaces().Get(context.Background(), "kube-system", metav1.GetOptions{}); err != nil {
+		return xerrors.Errorf("get kube-system namespace: %w", err)
+	}
```

Comment:
How about using [Forever](https://pkg.go.dev/k8s.io/apimachinery@v0.29.1/pkg/util/wait#Forever) ?

Reply:
Let's go with [`UntilWithContext`](https://pkg.go.dev/k8s.io/apimachinery/pkg/util/wait#UntilWithContext) so that it can be canceled. We can set the timeout with `context.WithDeadline`.

Reply:
Thanks. @sanposhiho